### ISSUE TITLE
How should "expected failure" decrypt results be formatted?

### ISSUE THREAD

**Issue by Kritner** [usnistgov/ACVP#28]
On a related note to #27 - some test vectors generated will have expected failures.  A failure (in the example of AES-GCM) could be an invalid tag for the operation.  In this case the original vector/tcId could look like:

```
{
    "tcId": 9999,
    "key": "FEFFE9928665731C6D6A8F9467308308",
    "iv": "CAFEBABEFACEDBADDECAF888",
    "ct": "42831EC2217774244B7221B784D0D49CE3AA212F2C02A4E035C17E2329ACA12E21D514B25466931C7D8F6A5AAC84AA051BA30B396A0AAC973D58E091473F5985",
    "aad": "3AD77BB40D7A3660A89ECAF32466EF97F5D3D58503B9699DE785895A96FDBAAF43B1CD7F598ECE23881B00E3ED0306887B0C785E27E8AD3F8223207104725DD4",
    "tag": "64C0232904AF398A5B67C10BD4205A35",  
},
```

Where the actual tag should be `64C0232904AF398A5B67C10B53A5024D` (last 8 characters of the tag were reversed in the test case.)

In this case, we would expect the client to report a failure (in some manner) for the decryption operation, as the `tag` will be unverifiable.

How will a decryption result of "failure" be represented in the `result` JSON?

tag sent as "FAIL":

```
{
    "tcId": 9999,
    "iv": "CAFEBABEFACEDBADDECAF888",
    "pt": "D9313225F88406E5A55909C5AFF5269A86A7A9531534F7DA2E4C303D8A318A721C3C0C95956809532FCF0E2449A6B525B16AEDF5AA0DE657BA637B391AAFD255",
    "tag": "FAIL"
}
```

new property "success" or similar `result`:

```
{
    "tcId": 9999,
    "iv": "CAFEBABEFACEDBADDECAF888",
    "pt": "D9313225F88406E5A55909C5AFF5269A86A7A9531534F7DA2E4C303D8A318A721C3C0C95956809532FCF0E2449A6B525B16AEDF5AA0DE657BA637B391AAFD255",
    "tag": "64C0232904AF398A5B67C10BD4205A35",
    "failureTest" : true
}
```

Other possibilities?

**Comment by Kritner** [comment]
@bfussell wrote: 

> The failure decrypt results could simply return:
"pt" : "FAIL"
or if we want a separate keyword for the failing case:
"result" : "FAIL"

We just had a quick meeting so I can at least proceed on my end, would you be opposed to having:

{
    "tcId": 9999,
    "pt": "D9313225F88406E5A55909C5AFF5269A86A7A9531534F7DA2E4C303D8A318A721C3C0C95956809532FCF0E2449A6B525B16AEDF5AA0DE657BA637B391AAFD255",
    "decryptFail" : true
}

to indicate an expected test failure as described in this issue?

**Comment by Kritner** [comment]
As per the call yesterday, expected failures will not contain the pt in the result.  Expected failure decrypt result could look like:
```
{
    "tcId": 9999,
    "decryptFail" : true
}
```

**Comment by atvassilev** [comment]
Addressed by PR #36.
